Source/Core/DolphinQt/Config/SettingsWindow.cpp

```diff
@@ -19,45 +19,134 @@
+          "QListWidget::item:selected { background: palette(highlight); "
+#if defined(_WIN32)
+          // Prevent text color change on focus loss on Windows.
+          // This seems to breaks the nice white text on macOS.
+          "color: palette(highlighted-text); "
+#endif
+          "} "
+          // Restore hovered item color, though not really the correct color.
+          "QListWidget::item:hover:!selected { background: %3; } "
+#endif
+#if defined(_WIN32)
+          // Remove ugly dotted outline on selected row.
+          "* { outline: none; } "
+#endif
+          )
+          .arg(QString::fromUtf8(list_background))
+          .arg(list_item_padding)
+          .arg(QString::fromUtf8(list_hover_background)));
+
+  layout->addWidget(m_navigation_list);
+
+  auto* const right_side = new QVBoxLayout;
+  layout->addLayout(right_side);
+
+  m_stacked_panes = new QStackedWidget;
+
+  right_side->addWidget(m_stacked_panes);
+
+  // The QFrame gives us some padding around the button.
+  auto* const button_frame = new QFrame;
+  auto* const button_layout = new QGridLayout{button_frame};
+  auto* const button_box = new QDialogButtonBox(QDialogButtonBox::Close);
+  right_side->addWidget(button_frame);
+  button_layout->addWidget(button_box);
 
-  // Add content to layout before dialog buttons.
-  m_tab_widget = new QTabWidget();
-  layout->addWidget(m_tab_widget);
+  connect(button_box, &QDialogButtonBox::rejected, this, &QDialog::reject);
 
-  m_tab_widget->addTab(GetWrappedWidget(new GeneralPane, this, 125, 100), tr("General"));
-  m_tab_widget->addTab(GetWrappedWidget(new InterfacePane, this, 125, 100), tr("Interface"));
-  m_tab_widget->addTab(GetWrappedWidget(new AudioPane, this, 125, 100), tr("Audio"));
-  m_tab_widget->addTab(GetWrappedWidget(new PathPane, this, 125, 100), tr("Paths"));
-  m_tab_widget->addTab(GetWrappedWidget(new GameCubePane, this, 125, 100), tr("GameCube"));
-  m_tab_widget->addTab(GetWrappedWidget(new WiiPane, this, 125, 100), tr("Wii"));
-  m_tab_widget->addTab(GetWrappedWidget(new AdvancedPane, this, 125, 200), tr("Advanced"));
+  connect(m_navigation_list, &QListWidget::currentRowChanged, m_stacked_panes,
+          &QStackedWidget::setCurrentIndex);
+}
+
+SettingsWindow::SettingsWindow(QWidget* parent) : StackedSettingsWindow{parent}
+{
+  setWindowTitle(tr("Settings"));
+
+  AddWrappedPane(new GeneralPane, tr("General"));
+  AddWrappedPane(new InterfacePane, tr("Interface"));
+  AddWrappedPane(new AudioPane, tr("Audio"));
+  AddWrappedPane(new PathPane, tr("Paths"));
+  AddWrappedPane(new GameCubePane, tr("GameCube"));
+  AddWrappedPane(new WiiPane, tr("Wii"));
+  AddWrappedPane(new AdvancedPane, tr("Advanced"));
+
+  OnDoneCreatingPanes();
+}
 
-  // Dialog box buttons
-  QDialogButtonBox* close_box = new QDialogButtonBox(QDialogButtonBox::Close);
+void StackedSettingsWindow::OnDoneCreatingPanes()
+{
+  // Make sure the first item is actually selected by default.
+  ActivatePane(0);
```

Comment:
Any reason you didn't do `SelectGeneralPane()` instead?  Feels like that would be more obvious than someone going to look and see what page 0 was.

Reply:
This is the base-class, `StackedSettingsWindow`.
`SelectGeneralPane` is a function of the derived-class, `SettingsWindow`.

`ActivatePane(0);` is a general way of activating the first pane and works for the main settings window and the game properties window.